pages/api/mock/[...slug].ts

```diff
@@ -253,26 +256,40 @@ export async function getMockedProperties(
   orgInfo?: {
     name: string;
     description: string;
-  }
+  },
+  isArray: boolean = false
 ): Promise<Record<string, any>> {
   const chunks = jsonSplitter(properties);
-  const transformed = transformProperties(chunks);
-  const primitiveOnly = Object.entries(transformed)
-    .filter(
-      ([_, value]) =>
-        value.type?.toLowerCase() !== "object" &&
-        value.type?.toLowerCase() !== "array"
-    )
-    .reduce((acc, [key, value]) => ({ ...acc, [key]: value }), {});
+  const allProperties = chunksToProperties(chunks);
+  const hardCodedProperties: Properties = {};
+  const propertiesForAi: Properties = {};
+
+  for (const [key, value] of Object.entries(allProperties)) {
+    const type = value.type?.toLowerCase() ?? "";
+    if (["object", "array"].includes(type)) continue;
+    if (type === "boolean") {
+      value.data = isArray
+        ? Array.from({ length: 3 }, () => Math.random() >= 0.5)
+        : Math.random() >= 0.5;
+      hardCodedProperties[key] = {
+        data: value.data,
+        type: "boolean",
+        path: value.path,
+      };
+    } else {
+      propertiesForAi[key] = value;
+    }
+  }
```

Comment:
Cheeky comment saying why some are hardcoded and others not might be useful :+1: